R/zchunk_LB110.For_FAO_R_Y.R

```diff
@@ -32,37 +34,116 @@ module_aglu_LB110.For_FAO_R_Y_DISABLED <- function(command, ...) {
+    # this tibble will be summed over by flow later, calculating export-import to give net export
+    L110.FAO_For_ALL_m3 <- rbind( L100.FAO_For_Prod_m3, L100.FAO_For_Imp_m3, L100.FAO_For_Exp_m3 )
+
+
+    # Lines 41-60 in original file
+    # Add regionID lookup vectors and GCAM_commodity label to the For_ALL tibble
+    # convert units and aggregate over flow to form Production and Net Export in units of bm3
+    # Use spread to have columns for Prod and NetExp instead of flow and value
+    # Add a column for the variable consumption, Cons=Prod-NetExp
+    L110.FAO_For_ALL_m3 %>%                                   # take the combined tibble
+      # do a left join on For_ALL tibble, match up the iso labels from the iso tibble,
+      #   This appends to the For_ALL tibble all of the information from the iso_GCAM_regID, including the column we actually
+      #   want, GCAM_region_ID. This column is all that we save:
+      mutate(GCAM_region_ID =  left_join( L110.FAO_For_ALL_m3, iso_GCAM_regID, by = c("iso"))$GCAM_region_ID ) %>%
+      #
+      mutate(GCAM_commodity = "Forest") %>%                   # add the forest commodity label
+      mutate(value = CONV_M3_BM3*value) %>%                   # convert the value units from m3 to bm3, had to add this constant to constants.R
+      mutate(flow = sub( "_m3", "_bm3", flow ) )  %>%         # update the labels in flow to reflect the new units of bm3
+      #
+      #we don't care about other identifiers anymore, only region id, commodity, flow and year corresponding to value:
+      select(GCAM_region_ID, GCAM_commodity, flow, year, value) %>%
+      #
+      # these two lines (group_by and summarise) compute Export-Import to give net exports, and leave the Prod alone,
+      #   for each commodity (just forest) in each GCAM region:
+      group_by(GCAM_region_ID, GCAM_commodity, flow,year) %>%
+      summarise(value = sum(value)) %>%
+      #
+      # build volumetric mass balance tibble:
+      #   instead of having flow and value for each year, region, commodity, we now want NetExp and Prod columns for each year, region, commodity.
+      #   The new DSR eliminates the use of L110.For_ALL_bm3_R_Y.prelim
+      spread(flow, value) %>%
+      #
+      mutate(Cons_bm3 = Prod_bm3 - NetExp_bm3) %>%                                     # form a new variable, consumption Cons = Prod-NetExp
+      select(GCAM_region_ID, GCAM_commodity, year, Prod_bm3, NetExp_bm3, Cons_bm3) ->  # reorder columns
+      L110.For_ALL_bm3_R_Y                                                             # save it in the region year R_Y tibble
+
+
+    # Lines 62 - 64 in original file
+    # Form Global values by summing over regions for each Prod, NetExp, Cons in tibble L110.For_ALL_bm3_R_Y;
+    # Then use the Global values to calculate the global consumption scaler that satisfies Global Production = Consumption
+    L110.For_ALL_bm3_R_Y %>%                     # take the region year R_Y tibble
```

Comment:
Maybe expand this comment to make it clear that we are prioritizing production data over consumption when there is a mismatch.